addons/l10n_mx/models/account_account.py

```diff
@@ -17,3 +1104,20 @@ def create(self, vals_list):
             tag_id = debit_tag.id if account.code[0] in DEBIT_CODES else credit_tag.id
             account.tag_ids = [Command.link(tag_id)]
         return accounts
+
+    @api.constrains('code', 'group_id')
+    def ensure_code_conforms_coa_sat(self):
+        incorrect_accounts = self.filtered(
+                lambda a: a.group_id and a.account_type != 'equity_unaffected' and a.company_id.country_code == 'MX'
+                and (a.code[:6] not in SAT_CODES or not THIRD_SECTION.fullmatch(a.code[6:]))
+        )
+        if incorrect_accounts:
+            account_names = ''.join(f'\n -   {a.name}' for a in incorrect_accounts)
+            raise RedirectWarning(
+                _("Some of your accounts do not respect the SAT code regulation.\n%s", account_names),
+                {
+                    'type': 'ir.actions.act_url',
+                    'url': f'https://www.odoo.com/{self.env.lang}/documentation/{version_info[0]}.0/applications/finance/fiscal_localizations/mexico.html#chart-of-accounts'
```

Comment:
`/{self.env.lang}/`  seems a little unnecessary, by default we should get either our current language on the website, or fallback to the browser language, or fallback to the default language (english I guess)

the `{version_info[0]}` also seems a little unnecessary but it should work for version other than master

we usually just put the major version (and master in master) then I guess someone use a script to update it when new major version is released, eg in 17.0: 1fb7571afbd0a914b7d818ea130055a280d00615